app/plugins/wamMaterialisedTaxonomy/wamMaterialisedTaxonomyPlugin.php

```diff
@@ -0,0 +1,119 @@
+require_once(__CA_LIB_DIR__."/core/Datamodel.php");
+
+/**
+ * The WAM Materialised Taxonomy plugin keeps copies of ancestral taxonomy ranks in child level names for easy access
+ */
+class wamMaterialisedTaxonomyPlugin extends BaseApplicationPlugin {
+
+	public function __construct($ps_plugin_path) {
+		parent::__construct();
+		$this->description = _t('Keeps copies of ancestral taxonomy ranks in child level names for easy access' );
+		$this->opo_config = Configuration::load($ps_plugin_path . DIRECTORY_SEPARATOR . 'conf' . DIRECTORY_SEPARATOR . 'materialisedTaxonomy.conf');
+	}
+
+	public function checkStatus() {
+		return array(
+			'description' => $this->getDescription(),
+			'errors' => array(),
+			'warnings' => array(),
+			'available' => true
+		);
+	}
+
+	/**
+	 * Give everybody access to this plugin
+	 *
+	 * @return array of actions that can be assigned to roles
+	 */
+	public static function getRoleActionList(){
+		return array(
+		);
+	}
+
+	/**
+	 * Iterates through the configured tables and sets proxy attribute values for fields where they have been mapped.
+	 *
+	 * @return true - required in order for subsequent hooks to fire
+	 */
+	public function hookPeriodicTask() {
+		$vb_skip_processed_rows = $this->opo_config->getBoolean('skip_processed_rows');
+		if ($va_table_config = $this->opo_config->getAssoc('tables')){
+			$vo_dm = Datamodel::load();
+			foreach($va_table_config as $vs_table_name => $va_type_info){
+				/** @var BundlableLabelableBaseModelWithAttributes $vo_table_instance */
+				$vo_table_instance = $vo_dm->getTableInstance($vs_table_name);
+				$vs_type_field = $vo_table_instance->getTypeFieldName();
+				foreach($va_type_info as $vs_type => $va_type_to_attribute_map){
+					$vn_type_id = $vo_table_instance->getTypeIDForCode($vs_type);
+					/** @var ca_list_items $vo_type_list_item */
+					$vo_type_list_item = new ca_list_items($vn_type_id);
+					$vo_child_types = $vo_type_list_item->getHierarchy(null, array('idsOnly' => true));
+					$vo_table_instance->getTypeList();
+					/** @var BaseSearchResult $vo_result */
+					$va_ids = $vo_table_instance->find(array($vs_type_field => $vo_child_types, 'deleted' => false), array('returnAs' => 'ids'));
+					foreach($va_ids as $vn_id){
+						$vo_table_instance->load($vn_id);
+						$va_new_values = array();
+						$va_ancestors = $vo_table_instance->getHierarchyAncestors($vn_id, array('idsOnly' => true, 'includeSelf' => true));
+						/**
+						 * @var int $vn_ancestor_id
+						 * @var BundlableLabelableBaseModelWithAttributes $vo_ancestor_instance
+						 */
+						foreach($va_ancestors as $vn_ancestor_id){
+							$vo_ancestor_instance = new $vs_table_name($vn_ancestor_id);
+							if($vs_ancestor_type = $vo_ancestor_instance->getTypeCode()){
+								if(isset($va_type_to_attribute_map[$vs_ancestor_type])){
+									// Get a label, but it's ok to use the cache
+									$vs_label = $vo_ancestor_instance->getLabelForDisplay(false);
```

Comment:
I think the `false` here will be implied if omitted, but no harm in being explicit.